app/src/main/java/com/example/dogbook/MainActivity/MainActivity.java

```diff
@@ -65,8 +68,19 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
             finish();
             return true;
         }
+        if (item.getItemId() == R.id.menuCompose) {
+            Intent intent = new Intent(this, ComposeActivity.class);
+            startActivityForResult(intent, REQUEST_CODE_COMPOSE_ACTIVITY);
+            return true;
+        }
         return super.onOptionsItemSelected(item);
     }
 
-
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == REQUEST_CODE_COMPOSE_ACTIVITY && resultCode == RESULT_OK && fragmentManager.findFragmentById(R.id.fragmentContainer) instanceof TimelineFragment) {
```

Comment:
Hm, this is getting to be a lot of conditions -- might want to pull the later check out into a method to increase readability that says something like `isTimelineFragment()`